### PR TITLE
Speed up torch engine w8a8 model init

### PR DESCRIPTION
**Pull request by yinfan98**
When using w8a8 model, it will cost a bunch of time for model loading. I found that when using 
```python
QRMSNorm.from_float()
## or
QLinear.from_float()
```
this func cal the per_channel_quant to get the scale and quanted weight, but it is useless. Because the para not load to model yet, it is just cal for all the zero... so I fix it when init the model on inference stage.

|      | time cost(s)           |
|--------------|------------------|
| cal the scale and weight         | 17 |
| dont cal         | 0.08 |

And I tested it on internLM and internLM2

### CODE REVIEW

**Comment by HIT-cwh** [comment]
Hi @yinfan98 !
Thank you for your contribution, we truly appreciate it. We are just a small step away from achieving success. For the next steps, please refer to [this document](https://github.com/open-mmlab/mmengine/blob/main/docs/zh_cn/notes/contributing.md#2-%E9%85%8D%E7%BD%AE-pre-commit) to configure the pre-commit then proceed to complete the lint check. We are looking forward to seeing the completion.

File: lmdeploy/pytorch/models/q_modules.py [hunk 1]
```diff
@@ -98,18 +101,23 @@ def __init__(self,
             self.register_parameter('bias', None)
 
     @classmethod
-    def from_float(cls, mod):
+    def from_float(cls, 
+                   mod: nn.Module,
+                   initialization: bool = True):
         """Class method to create a QLinear instance from a floating-point
-        module."""
+        module. initialization for dummy init."""
```
Comment (HIT-cwh) [inline lmdeploy/pytorch/models/q_modules.py:108]: It seems that `initialization = True` means real init and `initialization = False` means dummy init?

File: lmdeploy/pytorch/models/q_modules.py [hunk 2]
```diff
@@ -40,13 +40,17 @@ def __init__(self, hidden_size, eps=1e-6):
         self.variance_epsilon = eps
 
     @classmethod
-    def from_float(cls, mod):
+    def from_float(cls, mod: nn.Module, initialization: bool = True):
         """Class method to create a QRMSNorm instance from a floating-point
-        module."""
+        module.
+
+        initialization for dummy init.
```
Comment (HIT-cwh) [inline lmdeploy/pytorch/models/q_modules.py:47]:
```suggestion
        `initialization = True` for real init and initialization = False` for dummy init.
```

**Review by grimoire — APPROVED** [review]

**Comment by lvhan028** [comment]
Hi, @yinfan98 thanks very much for the contribution. 
Could you kindly provide the performance of w8a8?

**Comment by yinfan98** [comment]
@lvhan028 Sure, I will finish benchmark these days

**Pull request merged**